Fix intermittently failing test

We have an occasional test suite failure:

```sh
$ trial test.test_proteins.TestPathogenSampleFiles.testIdenticalReadsRemoved
[FAIL]
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/unittest/case.py", line 597, in run
    testMethod()
  File "/Users/terry/dark-matter/dark-matter/test/test_proteins.py", line 751, in testIdenticalReadsRemoved
    filename = pathogenSampleFiles.add('Lausannevirus', 'filename-1')
  File "/Users/terry/dark-matter/dark-matter/dark/proteins.py", line 144, in add
    for read in self._readsClass(proteinMatch['readsFilename']):
  File "/Users/terry/dark-matter/dark-matter/dark/reads.py", line 1122, in __iter__
    for read in subclassReads:
  File "/Users/terry/dark-matter/dark-matter/dark/fasta.py", line 116, in iter
    with asHandle(_file) as fp:
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/contextlib.py", line 59, in __enter__
    return next(self.gen)
  File "/Users/terry/dark-matter/dark-matter/dark/utils.py", line 127, in asHandle
    with open(fileNameOrHandle) as fp:
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/unittest/mock.py", line 917, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/unittest/mock.py", line 983, in _mock_call
    ret_val = effect(*args, **kwargs)
  File "/Users/terry/dark-matter/dark-matter/test/test_proteins.py", line 716, in sideEffect
    self.test.assertEqual('out/0.fasta', filename)
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/unittest/case.py", line 817, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/unittest/case.py", line 1190, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/unittest/case.py", line 662, in fail
    raise self.failureException(msg)
builtins.AssertionError: 'out/0.fasta' != 'out/1.fasta'
- out/0.fasta
?     ^
+ out/1.fasta
?     ^

test.test_proteins.TestPathogenSampleFiles.testIdenticalReadsRemoved
```

This will be due to keys coming out of a `dict` in a random order.

--- a/test/test_proteins.py
+++ b/test/test_proteins.py
@@ -709,27 +709,23 @@ class TestPathogenSampleFiles(TestCase):
             def __init__(self, test, manager):
                 self.test = test
                 self.manager = manager
-                self.count = 0
+                self.expectedFilenames = {'out/0.fasta', 'out/1.fasta',
+                                          'out/pathogen-0-sample-0.fasta'}
 
             def sideEffect(self, filename, *args, **kwargs):
-                if self.count == 0:
-                    self.test.assertEqual('out/0.fasta', filename)
-                    self.count += 1
-                    return File(['>id1\n', 'ACTG\n'])
-                elif self.count == 1:
-                    self.test.assertEqual('out/1.fasta', filename)
-                    self.count += 1
-                    return File(['>id1\n', 'ACTG\n', '>id2\n', 'CAGT\n'])
-                elif self.count == 2:
-                    self.test.assertEqual('out/pathogen-0-sample-0.fasta',
-                                          filename)
-                    self.count += 1
-                    return self.manager
-                else:
+                try:
+                    self.expectedFilenames.remove(filename)
+                except KeyError:
                     self.test.fail(
-                        'We are only supposed to be called 3 times. '
-                        'Filename: %r, Args: %r, Keyword args: %r.' %
-                        (filename, args, kwargs))
+                        'Open called with unexpected filename: %r, Args: %r, '
+                        'Keyword args: %r.' % (filename, args, kwargs))
+                else:
+                    if filename == 'out/0.fasta':
+                        return File(['>id1\n', 'ACTG\n'])
+                    elif filename == 'out/1.fasta':
+                        return File(['>id1\n', 'ACTG\n', '>id2\n', 'CAGT\n'])
+                    else:
+                        return self.manager
 
         fp = StringIO(
             '0.63 41.3 44.2 9 9 12 gi|327410| protein 77 [Lausannevirus]\n'
@@ -745,9 +741,12 @@ class TestPathogenSampleFiles(TestCase):
         pg.addFile('filename-1', fp)
         pathogenSampleFiles = PathogenSampleFiles(pg)
 
-        sideEffect = Open(self, manager()).sideEffect
+        opener = Open(self, manager())
         with patch.object(builtins, 'open') as mockMethod:
-            mockMethod.side_effect = sideEffect
+            mockMethod.side_effect = opener.sideEffect
             filename = pathogenSampleFiles.add('Lausannevirus', 'filename-1')
-            self.assertEqual('out/pathogen-0-sample-0.fasta', filename)
-            self.assertEqual('>id1\nACTG\n>id2\nCAGT\n', fastaIO.getvalue())
+
+        self.assertEqual('out/pathogen-0-sample-0.fasta', filename)
+        self.assertEqual('>id1\nACTG\n>id2\nCAGT\n', fastaIO.getvalue())
+        # Make sure all expected filenames were seen by the mocked open.
+        self.assertEqual(set(), opener.expectedFilenames)

